System/Random.hs

```diff
@@ -111,15 +109,11 @@ import Foreign.Ptr (plusPtr)
 import Foreign.Storable (peekByteOff, pokeByteOff)
 import GHC.ForeignPtr
 
-import Data.Char (chr, isSpace, ord)
 import Data.IORef (IORef, newIORef, readIORef, writeIORef, atomicModifyIORef')
-import Data.Ratio (denominator, numerator)
-import Data.Time (UTCTime(..), getCurrentTime)
-import System.CPUTime (getCPUTime)
 import System.IO.Unsafe (unsafePerformIO)
+import qualified System.Random.SplitMix as SM
 
-import GHC.Exts (Ptr(..), build, byteArrayContents#, unsafeCoerce#)
```

Comment:
Dropping is `byteArrayContents#, unsafeCoerce#` is incorrect here, since it is used with CPP here in `mutableByteArrayContentsCompat`. The fact that compiler gives you a warning is fixed in  #16

Reply:
I can confirm this

```
[1 of 1] Compiling System.Random    ( System/Random.hs, dist/build/System/Random.o )

System/Random.hs:123:10: error:
    • Variable not in scope:
        byteArrayContents# :: t0 -> ghc-prim-0.5.3:GHC.Prim.Addr#
    • Perhaps you meant ‘byteArrayContents’ (imported from Data.Primitive.ByteArray)
      Perhaps you want to add ‘byteArrayContents#’ to the import list
      in the import of ‘GHC.Exts’ (System/Random.hs:116:1-32).
    |
123 |   = Ptr (byteArrayContents# (unsafeCoerce# arr#))
    |          ^^^^^^^^^^^^^^^^^^

System/Random.hs:123:30: error:
    • Variable not in scope: unsafeCoerce# :: MutableByteArray# s -> t0
    • Perhaps you want to add ‘unsafeCoerce#’ to the import list
      in the import of ‘GHC.Exts’ (System/Random.hs:116:1-32).
    |
123 |   = Ptr (byteArrayContents# (unsafeCoerce# arr#))
    |                              ^^^^^^^^^^^^^
builder for '/nix/store/jmiqhkq4zicam8lr7snpd0d1msfakwp9-random-1.2.drv' failed with exit code 1
```

Reply:
:+1: fixed, thanks!